addons/account/models/account_move_line.py

```diff
@@ -1847,94 +1840,483 @@ def get_accounting_rate(vals):
         remaining_debit_amount_curr -= partial_debit_amount_currency
         remaining_credit_amount_curr += partial_credit_amount_currency
 
-        res['partial_vals'] = {
+        res['partial_values'] = {
             'amount': partial_amount,
             'debit_amount_currency': partial_debit_amount_currency,
             'credit_amount_currency': partial_credit_amount_currency,
-            'debit_move_id': debit_vals.get('record') and debit_vals['record'].id,
-            'credit_move_id': credit_vals.get('record') and credit_vals['record'].id,
+            'debit_move_id': debit_aml.id,
+            'credit_move_id': credit_aml.id,
         }
 
-        debit_vals['amount_residual'] = remaining_debit_amount
-        debit_vals['amount_residual_currency'] = remaining_debit_amount_curr
-        credit_vals['amount_residual'] = remaining_credit_amount
-        credit_vals['amount_residual_currency'] = remaining_credit_amount_curr
+        debit_values['amount_residual'] = remaining_debit_amount
+        debit_values['amount_residual_currency'] = remaining_debit_amount_curr
+        credit_values['amount_residual'] = remaining_credit_amount
+        credit_values['amount_residual_currency'] = remaining_credit_amount_curr
 
         if debit_fully_matched:
-            res['debit_vals'] = None
+            res['debit_values'] = None
         if credit_fully_matched:
-            res['credit_vals'] = None
+            res['credit_values'] = None
         return res
 
     @api.model
-    def _prepare_reconciliation_partials(self, vals_list):
+    def _prepare_reconciliation_amls(self, values_list):
         ''' Prepare the partials on the current journal items to perform the reconciliation.
         Note: The order of records in self is important because the journal items will be reconciled using this order.
-        :return: a tuple of 1) list of vals for partial reconciliation creation, 2) the list of vals for the exchange difference entries to be created
-        '''
-        debit_vals_list = iter([x for x in vals_list if x['balance'] > 0.0 or x['amount_currency'] > 0.0])
-        credit_vals_list = iter([x for x in vals_list if x['balance'] < 0.0 or x['amount_currency'] < 0.0])
-        debit_vals = None
-        credit_vals = None
 
-        partials_vals_list = []
-        exchange_data = {}
+        :return: a tuple of
+            1) list of vals for partial reconciliation creation,
+            2) the list of vals for the exchange difference entries to be created
+        '''
```

Comment:
That one for the end: It should be triple double quote :t-rex: :rofl:

Reply:
@Levizar How dare you!! :open_mouth: